src/Datadog.Trace.ClrProfiler.Native/module_metadata.h

```diff
@@ -45,15 +48,56 @@ class ModuleMetadata
         app_domain_id(app_domain_id),
         module_version_id(module_version_id),
         integrations(integrations),
-        corAssemblyProperty(corAssemblyProperty)
+        corAssemblyProperty(corAssemblyProperty),
+        integrations_cleanup(false)
     {
     }
+    ModuleMetadata(ComPtr<IMetaDataImport2> metadata_import, ComPtr<IMetaDataEmit2> metadata_emit,
+                   ComPtr<IMetaDataAssemblyImport> assembly_import, ComPtr<IMetaDataAssemblyEmit> assembly_emit,
+                   const WSTRING& assembly_name, const AppDomainID app_domain_id, const GUID module_version_id,
+                   const std::vector<IntegrationMethod> integrations, const AssemblyProperty* corAssemblyProperty) :
+        metadata_import(metadata_import),
+        metadata_emit(metadata_emit),
+        assembly_import(assembly_import),
+        assembly_emit(assembly_emit),
+        assemblyName(assembly_name),
+        app_domain_id(app_domain_id),
+        module_version_id(module_version_id),
+        integrations(new std::vector<trace::IntegrationMethod>(integrations)),
```

Comment:
I don't understand why you need to allocate a new object on this line, can you explain? From reading the other code, the `integrations` argument that's passed into the constructor is a new vector produced by `FilterIntegrationsByTarget` so it seems like it's not shared

Reply:
I do this because now we store a pointer in the module_metadata class, if the `FilterIntegrationsByTarget` creates the vector in the stack, the pointer will be invalid once the ModuleFinished callback returns. So to avoid this I force the creation in the heap and store that pointer to the heap. I have to check if the vector behave different and is already in the heap, but that's the reason is done this way.

Reply:
Same as Zach.
If you need a new instance, why using new ? why not using `std::unique_ptr`. It would simplify the dtor (no dtor at all)

Reply:
I didn't use `std::unique_ptr` in this case because the 1module metadata1 class behave differently depending if is CallTarget or CallSite. 

In the CallTarget case I want to use the already heap allocated integrations vector loaded in the Initialization without a copy and without calling the dtor or deleting the allocation, so in that case I just want a copy of the pointer. 

Meanwhile in CallSite we first filter the global integrations vector and return a new vector scoped to that module so... in that case I want to copy the vector to the heap and also call the dtor on finalization, because no other module we have the same copy.

Reply:
I made changes to remove the different behaviours of the module_metadata class in 79804dd, that allowed me to replace the pointers to `unique_ptr`